controllers/operatorpolicy_status.go

```diff
@@ -461,11 +480,98 @@ func buildDeploymentCond(
 	}
 }
 
-var noDeploymentsCond = metav1.Condition{
-	Type:    deploymentConditionType,
-	Status:  metav1.ConditionTrue,
-	Reason:  "NoRelevantDeployments",
-	Message: "The ClusterServiceVersion is missing, thus meaning there are no relevant deployments",
+func getCatalogCondAndObj(
+	catalogName string,
+	catalogNS string,
+	isUnhealthy bool,
+	isMissing bool,
+) (metav1.Condition, policyv1.RelatedObject) {
+	return catalogSourceFindCond(isUnhealthy, isMissing),
+		catalogSourceObj(catalogName, catalogNS, isUnhealthy, isMissing)
+}
```

Comment:
I think this function just adds an unnecessary extra layer when reading the code. I'd rather just have both of the functions in here used individually.